src/controller/stream-controller.js

```diff
@@ -230,34 +230,33 @@ class StreamController extends BaseStreamController {
       bufferEnd = bufferInfo.end,
       frag;
 
-    if (levelDetails.initSegment && !levelDetails.initSegment.data) {
-      frag = levelDetails.initSegment;
-    } else {
-      // in case of live playlist we need to ensure that requested position is not located before playlist start
-      if (levelDetails.live) {
-        let initialLiveManifestSize = this.config.initialLiveManifestSize;
-        if (fragLen < initialLiveManifestSize) {
-          logger.warn(`Can not start playback of a level, reason: not enough fragments ${fragLen} < ${initialLiveManifestSize}`);
-          return;
-        }
+    // in case of live playlist we need to ensure that requested position is not located before playlist start
+    if (levelDetails.live) {
+      let initialLiveManifestSize = this.config.initialLiveManifestSize;
+      if (fragLen < initialLiveManifestSize) {
+        logger.warn(`Can not start playback of a level, reason: not enough fragments ${fragLen} < ${initialLiveManifestSize}`);
+        return;
+      }
 
-        frag = this._ensureFragmentAtLivePoint(levelDetails, bufferEnd, start, end, fragPrevious, fragments, fragLen);
-        // if it explicitely returns null don't load any fragment and exit function now
-        if (frag === null) {
-          return;
-        }
-      } else {
-        // VoD playlist: if bufferEnd before start of playlist, load first fragment
-        if (bufferEnd < start) {
-          frag = fragments[0];
-        }
+      frag = this._ensureFragmentAtLivePoint(levelDetails, bufferEnd, start, end, fragPrevious, fragments, fragLen);
+      // if it explicitely returns null don't load any fragment and exit function now
```

Comment:
@gregrealeyes explicitely

Reply:
I fixed this, though I don't know why it isn't being reflected here like it does on Bitbucket